ipmof/interpenetration.py

```diff
@@ -412,3 +417,41 @@ def run_interpenetration(base_mof, mobile_mof, emap, atom_list, sim_par, sim_dir
                 joined_packed_mof.export(export_dir, file_format=sim_par['export_format'])
 
     export_interpenetration_results(sim_par, structure_info, summary, export_dir)
+
+
+def get_interpenetration_list(sim_par, sim_dir):
+    """
+    Returns a dictionary containing paths for energy map and mof files for each MOF combination.
+    Energy map is read from ~/energy_map and MOFs are read from ~/mof.
+    All possible combinations are then arranged in a dictionary format:
+        - If self_interpenetration parameter if False homo-interpenetration is excluded
+        - Reverse combinations are excluded (ex: if MOF1_MOF2 in list then MOF2_MOF1 is not selected)
+
+    Format: interpenetration_list = {'emap_path': [], 'emap_mof_path': [], 'ip_mof_path': []}
+    """
+    mof_path_list = get_mof_list(sim_par, sim_dir)
+    emap_path_list = os.listdir(sim_dir['energy_map_dir'])
+
+    interpenetration_list = {'emap_path': [], 'emap_mof_path': [], 'ip_mof_path': []}
+
+    for emap_path in emap_path_list:
+
+        emap_mof_name = os.path.basename(emap_path).split('_emap')[0]
+        # What if multiple files are returned?? (IndexError if file cannot be found)
+        emap_mof_path = glob(os.path.join(sim_dir['mof_dir'], emap_mof_name) + '*')[0]
+        emap_path = os.path.join(sim_dir['energy_map_dir'], emap_path)
+
+        for ip_mof_path in mof_path_list:
+
+            if emap_mof_path == ip_mof_path and not sim_par['self_interpenetration']:
+                continue
+
+            elif emap_mof_path in interpenetration_list['ip_mof_path'] and ip_mof_path in interpenetration_list['emap_mof_path']:
+                continue
+
+            else:
+                interpenetration_list['ip_mof_path'].append(ip_mof_path)
+                interpenetration_list['emap_mof_path'].append(emap_mof_path)
+                interpenetration_list['emap_path'].append(emap_path)
```

Comment:
I decided to keep the tuple format while passing the paths to run_interpenetration method.
I am still unpacking the individual paths in both the method and the main script that calls the method. I decided to do it that way so that the method is called with less number of arguments and the individual paths can be seen clearly.